core/include/drivers/gpio.h

```diff
@@ -40,4 +66,126 @@ struct gpio_ops {
 			      enum gpio_interrupt ena_dis);
 };
 
+struct gpio {
+	struct gpio_chip *chip;
+	uint32_t dt_flags;
+	unsigned int pin;
+};
+
+static inline bool gpio_ops_is_valid(const struct gpio_ops *ops)
+{
+	return ops->set_direction && ops->get_direction && ops->get_value &&
+	       ops->set_value;
+}
+
+static inline void gpio_set_direction(struct gpio *gpio, enum gpio_dir dir)
+{
+	gpio->chip->ops->set_direction(gpio->chip, gpio->pin, dir);
+}
+
+static inline enum gpio_dir gpio_get_direction(struct gpio *gpio)
+{
+	return gpio->chip->ops->get_direction(gpio->chip, gpio->pin);
+}
+
+static inline void gpio_set_value(struct gpio *gpio, enum gpio_level value)
+{
+	if (gpio->dt_flags & GPIO_ACTIVE_LOW)
+		value = !value;
```

Comment:
is this shoprt cut ok? or should we enforce `GPIO_LEVEL_LOW=0, GPIO_LEVEL_HIGH=1` at lines 41/42?